### ISSUE TITLE
Cubemap reflections don't align correctly.

### ISSUE THREAD

**Issue by slimbuck** [playcanvas/engine#2460]
It looks like cubemap reflections are sometimes incorrectly aligned. This may be related to box projection:

![Screenshot 2020-10-05 at 09 00 17](https://user-images.githubusercontent.com/11276292/95054125-50f03c00-06e9-11eb-9b3e-821c1bacb0b7.png)

**Comment by Maksims** [comment]
Also, I've noticed how different engines, project cubemap differently. You can see it in https://github.com/cx20/gltf-test check `Damaged Helmet` demos.

**Comment by raytranuk** [comment]
I took a look at this - and its partially caused by the source cube map not lining up exactly on cube corners (you can see this by switching from box to normal projection) - but mostly because the geometry doesn't line up well with the box used for the box projection - this is a common problem when using box projection with shapes that are not boxes.

**Comment by raytranuk** [comment]
Below is a list that IMO can be done to improve reflections in PlayCanvas (which can be discussed and prioritized against other work to establish at what point they can be delivered): 
1) Add support for multiple reflection probes and reflection probe blending
2) Add in-editor rendering of reflection probes
3) Add Screen Space reflection implementation
4) Add Ray Traced reflection implementation
We can update this issue if/when new reflection related issues/feature are planned.

**Comment by mvaligursky** [comment]
> Also, I've noticed how different engines, project cubemap differently. You can see it in https://github.com/cx20/gltf-test check `Damaged Helmet` demos.

agreed, I had a look just now .. it feels like the range of reflection gets clipped in playcanvas? All other engines render reflection almost white, but playcanvas renders it half grey .. feels similar to https://github.com/playcanvas/engine/pull/2019

**Comment by Maksims** [comment]
> agreed, I had a look just now .. it feels like the range of reflection gets clipped in playcanvas? All other engines render reflection almost white, but playcanvas renders it half grey .. feels similar to #2019

Not sure what going on with color there, but LDR cubemap definitely does lack of color depth.
But notice difference in projection, how it reflects. Different engines reflect cubemap differently.

**Comment by raytranuk** [comment]
I can certainly see the brightness of the reflection is much duller looking in PlayCanvas - this is something we should probably look to investigate/fix. It also looks like the helmets are not oriented the same relative to the cube map in all samples - so its difficult to see if projection is different, but I would be surprised if they don't just use the reflection vector as the coordinates to lookup the cube map - and calculate the reflection vector as view position to fragment position reflected by the fragment normal. If rotated cube maps are supported, the reflection vector would be rotated too.

**Comment by slimbuck** [comment]
@raytranuk is there any work planned here, what was the conclusion on this one?

**Comment by raytranuk** [comment]
@slimbuck - I added the 4 things I suggested for reflection improvements as feature requests - so we can assess when they should be tackled - and I was thinking about a prototype SSR implementation as part of my (currently paused) post process experiments.

**Comment by willeastcott** [comment]
Hey @slimbuck, with the recent overhaul of the reflection code, should we expect any improvements or change in behavior here?

**Comment by slimbuck** [comment]
Hi @willeastcott, no I wouldn't expect any significant change here.
